network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NodeBreakerViewImpl.java

```diff
@@ -125,44 +126,110 @@ public void traverse(int[] nodes, VoltageLevel.NodeBreakerView.TopologyTraverser
     public void traverse(int node, VoltageLevel.NodeBreakerView.TopologyTraverser traverser) {
         Objects.requireNonNull(traverser);
         checkBusBreakerTopology();
-        traverseFromNode(node, traverser);
+        traverseFromNode(node, TraversalType.DEPTH_FIRST, traverser);
     }
 
-    boolean traverseFromNode(int node, VoltageLevel.NodeBreakerView.TopologyTraverser traverser) {
+    boolean traverseFromNode(int node, TraversalType traversalType, VoltageLevel.NodeBreakerView.TopologyTraverser traverser) {
         Graph<Integer, Edge> graph = NodeBreakerTopology.INSTANCE.buildGraph(index, getVoltageLevelResource(), true, true);
         Set<Integer> done = new HashSet<>();
-        return traverseFromNode(graph, node, traverser, done);
+        return traverseFromNode(graph, node, traversalType, traverser, done);
     }
 
-    private boolean traverseFromNode(Graph<Integer, Edge> graph, int node, VoltageLevel.NodeBreakerView.TopologyTraverser traverser,
+    private boolean traverseFromNode(Graph<Integer, Edge> graph, int node, TraversalType traversalType, VoltageLevel.NodeBreakerView.TopologyTraverser traverser,
                                      Set<Integer> done) {
-        if (done.contains(node)) {
-            return true;
-        }
-        done.add(node);
-
-        for (Edge edge : graph.edgesOf(node)) {
-            NodeBreakerBiConnectable biConnectable = edge.getBiConnectable();
-            int nextNode = biConnectable.getNode1() == node ? biConnectable.getNode2() : biConnectable.getNode1();
-            TraverseResult result;
-            if (done.contains(nextNode)) {
-                continue;
-            }
-            if (biConnectable instanceof SwitchAttributes) {
-                result = traverseSwitch(traverser, biConnectable, node, nextNode);
-            } else if (biConnectable instanceof InternalConnectionAttributes) {
-                result = traverser.traverse(node, null, nextNode);
-            } else {
-                throw new AssertionError();
+        if (traversalType == TraversalType.DEPTH_FIRST) {   // traversal by depth first
```

Comment:
Separate DFS and BFS in two differents methods, otherwise it's a bit difficult to differentiate the recursive DFS with the BFS